src/packages/hex.gleam

```diff
@@ -3,41 +3,204 @@ import gleam/json
 import gleam/http
 import gleam/result
 import gleam/hackney
+import gleam/list
 import gleam/dynamic.{Decoder} as d
+import time/parse.{parse_iso8601_to_gregorian_seconds}
+import gleam/pgo
+import gleam/string
+import gleam/int
+import gleam/option.{Option}
 
 pub type Error {
   HackneyError(hackney.Error)
   JsonError(json.DecodeError)
+  DatabaseError(pgo.QueryError)
+  ItemNotInListError(Nil)
 }
 
-pub fn query() {
+pub fn query(db: pgo.Connection) {
+  try last_scanned = get_last_scanned(db)
+
+  try packages = query_all_packages([], 1, last_scanned)
+
+  // Only update last scanned at if packages were actually
+  // found that way we only ever use the time from
+  // Hex's API 
+  case packages
+  |> list.length >= 1 {
+    True -> {
+      let len =
+        packages
+        |> list.length
+      assert Ok(last_package) =
+        packages
+        |> list.at(len - 1)
```

Comment:
There's `list.last`, but I think it would write the code so that it gets the one that is latest without depending on the order. This will be more reliable and easier to understand for the reader.

Reply:
removed by implementing a change elsewhere